File: app/src/main/java/com/quran/labs/androidquran/widgets/WidgetService.java
package com.quran.labs.androidquran.widgets;

import android.appwidget.AppWidgetManager;
import android.content.Intent;
import android.widget.RemoteViewsService;


public class WidgetService extends RemoteViewsService {
/*
* So pretty simple just defining the Adapter of the listview
* here Adapter is ListProvider
* */

  @Override
  public RemoteViewsFactory onGetViewFactory(Intent intent) {
    int appWidgetId = intent.getIntExtra(
        AppWidgetManager.EXTRA_APPWIDGET_ID,
        AppWidgetManager.INVALID_APPWIDGET_ID);
    return (new WidgetListProvider(this.getApplicationContext(), intent));
  }
}

File: app/src/main/java/com/quran/labs/androidquran/widgets/WidgetService.kt
package com.quran.labs.androidquran.widgets

import android.content.Intent
import android.widget.RemoteViewsService

class WidgetService : RemoteViewsService() {
  /*
* So pretty simple just defining the Adapter of the listview
* here Adapter is ListProvider
* */
  override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
    return WidgetListProvider(applicationContext)
  }
}
